testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/util/ProjectTestsUtils.java

```diff
@@ -360,42 +370,80 @@ public static void waitForUpdateEditorJob() {
 		}
 	}
 
+	/** Delegates to {@link #waitForAllJobs(long)} with default values {@link #MAX_WAIT_2_MINUTES}. */
+	public static void waitForAllJobs() {
+		waitForAllJobs(MAX_WAIT_2_MINUTES);
+	}
+
 	/**
 	 * Waits for all the jobs that have status {@link Job#RUNNING} of {@link Job#WAITING}. Uses {@link Thread#sleep}, so
 	 * use with care.
+	 *
+	 * Intentionally throws runtime exception if there are still jobs running after given timeout. Don't catch it,
+	 * increase timeout in your tests.
+	 *
+	 * @param maxWait
+	 *            maximum wait time in {@link TimeUnit#MILLISECONDS}
 	 */
-	public static void waitForAllJobs() {
-		int maxWait = 100 * 60 * 2;
-		long start = System.currentTimeMillis();
-		long end = start;
+	public static void waitForAllJobs(final long maxWait) {
+		if (maxWait < 1)
+			throw new IllegalArgumentException("Wait time needs to be > 0, was " + maxWait + ".");
+
+		final boolean runsInUI = UIUtils.runsInUIThread();
+		if (runsInUI)
+			LOGGER.warn("Waiting for jobs runs in the UI thread which can lead to UI thread starvation.");
+
+		List<String> foundJobs = listJobsRunningWaiting();
+		if (foundJobs.isEmpty()) {
+			LOGGER.info("No running nor waiting jobs found, maybe all have already finished.");
+			return;
+		}
 		boolean wasInterrupted = false;
+		boolean wasCancelled = false;
 		boolean foundJob = false;
+		Stopwatch sw = Stopwatch.createStarted();
 		do {
 			try {
-				List<String> foundJobs = listJobsRunningWaiting();
-				if (!foundJobs.isEmpty()) {
-					foundJob = true;
-					// Job.getJobManager().join(null, null);
-					Thread.sleep(100L);
+				foundJobs = listJobsRunningWaiting();
+				foundJob = !foundJobs.isEmpty();
+				if (foundJob) {
+					if (LOGGER.isInfoEnabled())
+						LOGGER.info("Found " + foundJobs.size() + " after " + sw + ", going to sleep for a while.");
+
+					if (runsInUI)
+						UIUtils.waitForUiThread();
+					else
+						Thread.sleep(CHECK_INTERVAL_100_MS);
 				}
-				wasInterrupted = false;
-				end = System.currentTimeMillis();
 			} catch (OperationCanceledException e) {
-				e.printStackTrace();
+				wasCancelled = true;
+				LOGGER.error("Waiting for jobs was cancelled after " + sw + ".", e);
 			} catch (InterruptedException e) {
 				wasInterrupted = true;
+				LOGGER.error("Waiting for jobs was interrupted after " + sw + ".", e);
 			}
-		} while (wasInterrupted && (end - start) < maxWait);
-		if (!foundJob) {
-			LOGGER.debug("No running nor waiting jobs found, maybe all have already finished.");
+		} while (sw.elapsed(TimeUnit.MILLISECONDS) < maxWait
+				&& foundJob == true
+				&& wasInterrupted == false
+				&& wasCancelled == false);
```

Comment:
why check a boolean for equals? (booleanVar == false/true)

Reply:
somehow I find it more readable then 
```java
		} while (sw.elapsed(TimeUnit.MILLISECONDS) < maxWait
				&& foundJob
				&& !wasInterrupted
				&& !wasCancelled);
```